docs/guide/acquiring_data/helioviewer.rst

```diff
@@ -147,20 +195,20 @@ Map object.
 
 The overall syntax is similar to the *download_png* request, expect instead of
 specifying a single string to indicate which layers to use, here we
-can specify the values as separate keyword arguments.::
+specify the sourceId of the image we want to download.::
 
    >>> from sunpy.net.helioviewer import HelioviewerClient
    >>> import matplotlib.pyplot as plt
    >>> from astropy.units import Quantity
    >>> from sunpy.map import Map
    >>> hv = HelioviewerClient()  # doctest: +REMOTE_DATA
-   >>> filepath = hv.download_jp2('2012/07/05 00:30:00', observatory='SDO', instrument='HMI', detector='HMI', measurement='continuum')  # doctest: +REMOTE_DATA
+   >>> data_sources = hv.get_data_sources()  # doctest: +REMOTE_DATA
+   >>> filepath = hv.download_jp2('2012/07/05 00:30:00', sourceid=data_sources['SDO']['HMI']['continuum']['sourceId'])  # doctest: +REMOTE_DATA
```

Comment:
Version 2 of Helioviewer API provides more features than version 1. But in this version, a lot of things have changed as compared to the older version. Earlier source_id was an optional parameter and user can download the image even by passing parameters like observatory, instruments etc but now without source_id you cannot download the image. So source_id became a necessary parameter from optional one. So we had to remove even the tests that were testing this API on the basis of parameters as using source_id is mandatory now.

Reply:
So what we want is that we incorporate the functionality of the older API of using observatory='SDO', instrument='HMI', detector='HMI', measurement='continuum'
in the newer one?

Reply:
Yes.

Reply:
I am not sure if I will be able to do it, but yes I will surely give it a try

Reply:
Won't we be having problem were we have all 4 i.e 
Observatory, instrument, detector and measurement?

Reply:
No. If the word doesn't exist for it we want to use `None` and have the sourceID be returned correctly.

Reply:
Um, just a small doubt.
The observatory, instruments etc will be users choice right?
I mean what we can do is we can ask them to input all these before hand and then just give them the source_id as per the entries they do

Reply:
We will allow the users to enter the names, if they are wrong, it will error and we can return the message saying so. 

We could create a function that returns the source id given those inputs as you say. 
That might be the easier way forward.

Reply:
I have written a function for it. Please let me know if it has any issues

Reply:
What happened to the commit history?

Reply:
Has the commit history changed?

Reply:
I forced push my commit. I guess that ruined the history?

Reply:
Yeah, you needed to update your history locally and then commit on top. 
I don't think it affected the actual code changes tho, so its fine.

Reply:
It was giving merging conflicts when I was trying to pull this

Reply:
Yeah, I had rebased the PR on top of master which had some changes. You generally can't merge in a rebase, you have to just overwrite the local branch.

Reply:
Oh ok ok. 
By the way is the function ok?

Reply:
It's not a bad start but I don't think we made it very clear what we are after.

We want the function to construct and return a dictionary full of the all the source Ids that someone could then index to return the source id. Then a user or a function could access the dictionary using that tuple. 

The function just creates a dictionary and then clears it, you could delete that part and nothing would change. Also, the for loop overwrites the source id?

Reply:
So what we can do is that we can make a dictionary containing the tuples as the key and corresponding source ID as the value. Whenever the user enters the key we will return the source ID?

It does not overwrite the source id. It just returns the value of the key in the dictionary

Reply:
Yes.

Reply:
> It's not a bad start but I don't think we made it very clear what we are after.
> 
> We want the function to construct and return a dictionary full of the all the source Ids that someone could then index to return the source id. Then a user or a function could access the dictionary using that tuple.
> 
> The function just creates a dictionary and then clears it, you could delete that part and nothing would change. Also, the for loop overwrites the source id?

Is it not better that the function creates a dictionary but returns the sourceID as per the tuple rather than returning the dictionary?

Reply:
For creating the dictionary we can reuse the code written in the rst file of helioviewer documentation

Reply:
Well the function as it stands creates a dictionary that serves no purpose. 

We want the user to be able to print out the entire list to see what is available. Once they know what they want, they can either then pass in the sourceid or pass in the older API.

Reply:
Um, check this one. Does this satisfy our needs?

Reply:
Its much closer but there a few issues.
It returns a string instead of a dictionary. 
It seems to split some of the keywords like `Open/Al_mesh`. I'm not sure if they are meant to be like that.

Reply:
Oh I guess that's because of the way I'm returning the dictionary. I'll change it so that it returns a dictionary.